src/app/auth/services/auth.service.ts

```diff
@@ -1,32 +1,111 @@
-import { Injectable } from "@angular/core";
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, catchError, map, Observable, of } from 'rxjs';
+import { SessionStorageService } from './session-storage.service';
+import { APIResult, LoginResponse, LoginUser, LogoutResponse, ResgistrationResponse, User } from '../auth.models';
+import { ADMIN_EMAIL, UserStoreService } from '@app/user/services/user-store.service';
 
 @Injectable({
-  providedIn: "root",
+  providedIn: 'root',
 })
 export class AuthService {
-  login(user: any) {
-    // replace 'any' with the required interface
-    // Add your code here
+  private isAuthorized$$ = new BehaviorSubject<boolean>(false);
+  public isAuthorized$: Observable<boolean> = this.isAuthorized$$.asObservable();
+
+  constructor(
+    private http: HttpClient,
+    private sessionStorage: SessionStorageService,
+    private userStore: UserStoreService
+  ) {}
+
+  login(user: LoginUser): Observable<APIResult> {
+    return this.http.post<LoginResponse>(this.getLoginUrl(), user).pipe(
+      map(response => {
+        const token = response?.result;
+        if (token) {
+          this.sessionStorage.setToken(token);
+          this.isAuthorized$$.next(true);
+          if (response.user.email === ADMIN_EMAIL) {
```

Comment:
Well, probably we don't want to keep track of all admins in out client applications, especially not hardcoding them in the source code. What if admin's email changes? Or another admin is added? Yeah... It would require some serious code changes along with the releases.
So, please disregard the values here and specifically do not assign it to a variable. But you can always call the API to get the user details where you can easily check the role and name properties.

Reply:
Ohh nice! I missed that admin role is coming in the me API. For me it would be logical it is already in login response ... this is why I did not search ... . But I see now the concept. 
Just a question, is it normal that in a response a password field comes back like here?
Done. me APi for getting user is called right after successful login and admin is deetermined based on role.